allennlp/semparse/domain_languages/domain_language.py

```diff
@@ -258,6 +277,26 @@ def execute(self, logical_form: str):
         expression = util.lisp_to_nested_expression(logical_form)
         return self._execute_expression(expression)
 
+    def execute_action_sequence(self, action_sequence: List[str], side_arguments: List[Dict] = None):
+        """
+        Executes the program defined by an action sequence directly, without needing the overhead
+        of translating to a logical form first.  For any given program, :func:`execute` and this
+        function are equivalent, they just take different representations of the program, so you
+        can use whichever is more efficient.
+
+        Also, if you have state or side arguments associated with particular production rules
+        (e.g., the decoder's attention on an input utterance when a predicate was predicted), you
+        `must` use this function to execute the logical form, instead of :func:`execute`, so that
```

Comment:
Since `execute` won't work with side arguments, can you also add some logic in `execute` to identify if any of the predicates expects side arguments and throw an error?

Reply:
You will already get an `ExecutionError` that says there aren't enough arguments passed to the function.  I'm not sure we need to make that logic more complex just to give a slightly improved error.  If it comes up with people getting confused by that, we can consider fixing it.